Imagine/Cache/CacheManager.php

```diff
@@ -109,12 +117,12 @@ public function getBrowserPath($path, $filter, array $runtimeConfig = [], $resol
         if (!empty($runtimeConfig)) {
             $rcPath = $this->getRuntimePath($path, $runtimeConfig);
 
-            return $this->isStored($rcPath, $filter, $resolver) ?
+            return !$this->webpGenerate && $this->isStored($rcPath, $filter, $resolver) ?
                 $this->resolve($rcPath, $filter, $resolver) :
                 $this->generateUrl($path, $filter, $runtimeConfig, $resolver, $referenceType);
         }
 
-        return $this->isStored($path, $filter, $resolver) ?
+        return !$this->webpGenerate && $this->isStored($path, $filter, $resolver) ?
```

Comment:
This can’t be the solution as when WebP is enabled, all URLs will go through the controller. This work in your project but doesn’t with this bundle and won’t work with s3, etc...

Reply:
Why won't it work? The request goes through the controller and is cached by the client.  A repeated call by the same client will not reach the controller and will be immediately redirected by the browser to s3, etc.

Reply:
If you serve the image to 1M clients, you will get 1M requests to the Controller.

Previously, only the first request goes through the Controller. The 999,999 others are directly served from s3.

Reply:
Yes, but we receive 1M requests to s3 from the server to check for the existence of the file. This is also not a free operation. This affects the page rendering speed. We can move the LiipImagine controller to a separate endpoint and better manage the load.

Reply:
We can improve performance by caching the full page as static, but then we definitely cannot resolve WebP in templates.
We can introduce one more option in the configurations that will allow us to resolve WebP in templates, but there can be much more unexpected problems with this.

Reply:
> Yes, but we receive 1M requests to s3 from the server to check for the existence of the file. This is also not a free operation.

This is not free, but it's what happen when WebP support is disabled. Having the same behavior is a must have IMHO.

> We can improve performance by caching the full page as static, but then we definitely cannot resolve WebP in templates.

Page caching is out of scope of this bundle.

> We can introduce one more option in the configurations that will allow us to resolve WebP in templates, but there can be much more unexpected problems with this.

I don't think you need more configuration options. As you do content negotiation in the controller, you should do the same in the twig extension.

There is probably some logic to extract from the `FilterService` (you can't use it directly because it would generate the cached file when the twig is rendered).

Reply:
> Page caching is out of scope of this bundle.

That is why i say that we cannot resolve WebP in templates, since we do not know how these templates will be used. And we cannot do this:

> This is not free, but it's what happen when WebP support is disabled. Having the same behavior is a must have IMHO.

Reply:
There are many ways to avoid calling a controller, but they all require individual server configuration. Resolving in the controller is an easy way to make WebP support out of the box for everyone, including shared hosting.

This is a simple example that saves you the trouble of accessing the controller:

```
map $http_accept $webp_suffix {
    "~*webp"  ".webp";
}

server {

   # ...

    location ~ \.(png|jpe?g)(\?[a-f0-9]+)?$ {
        expires 365d;
        access_log off;
        add_header Cache-Control "public";
        try_files $uri$webp_suffix $uri /index.php$is_args$args;
    }

   # ...
}
```

> *I hope i take the time to describe in detail how users can improve the rendering speed of pages and images when using [Amazon S3](https://symfony.com/doc/2.x/bundles/LiipImagineBundle/cache-resolver/amazons3.html) or [file system](https://symfony.com/doc/2.x/bundles/LiipImagineBundle/cache-resolver/web_path.html) as cache.*